Fix SyntaxError: Illegal return statement

gitim opened the pull request:
Faced with SyntaxError in this package, because some package in my project depend from this package.

ljharb reviewed (changes requested):
See https://github.com/inspect-js/is-set/pull/4#pullrequestreview-332658947

This is not a syntax error whatsoever; this is a node module and top level return is allowed in it.

What bundler are you using that breaks on valid node modules?

ljharb commented:
https://github.com/inspect-js/is-set/pull/4#pullrequestreview-332658947

Jessidhia commented on index.js:
```diff
@@ -20,7 +19,6 @@ if (!$mapHas) {
 		// `Map` does not have a `has` method
 		return false;
 	};
-	return;
```
Even if you were to get rid of these `return`s, this is not it; you need to convert the rest of the file into `else` bodies.

ljharb committed:
[Refactor] avoid top-level return, because babel and webpack are broken

Closes #5. Closes #4. Closes #3.

Closes https://github.com/inspect-js/node-deep-equal/issues/78.
Closes https://github.com/es-shims/Promise.allSettled/issues/7.
Closes https://github.com/airbnb/js-shims/issues/12.

Relates to https://github.com/inspect-js/is-set/pull/4, https://github.com/inspect-js/is-set/pull/5

Addresses https://github.com/storybookjs/storybook/issues/9154.
